src/PersistenceProviders/Proto.Persistence.Sqlite/SqliteProvider.cs

```diff
@@ -147,13 +215,25 @@ public Task<(object Snapshot, long Index)> GetSnapshotAsync(string actorName)
         {
             try
             {
-                using (var db = new SqlitePersistenceContext(_datasource))
-                {
-                    var item = new Snapshot(actorName, index, JsonConvert.SerializeObject(snapshot, new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.All }));
+                var item = new Snapshot(actorName, index, JsonConvert.SerializeObject(snapshot, new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.All }));
 
-                    await db.Snapshots.AddAsync(item);
+                using (var connection = new SqliteConnection(_connectionString))
+                {
+                    connection.Open();
 
-                    await db.SaveChangesAsync();
+                    using (var transaction = connection.BeginTransaction())
```

Comment:
Don't need a transaction over a single command.  Sqlite implicitly has a transaction on this.